fw/hpack.c

```diff
@@ -232,7 +232,7 @@ do {								\
 			goto out;				\
 		}						\
 		__c = *src++;					\
-		x = (__c & 127) << __m | x;			\
+		x += (__c & 127) << __m;			\
```

Comment:
I believe CI does execute the unit tests, but for my build (kernel and tempesta as of the review date) I see many failures, so could you please check the tests?
```
# make clean test
....
[  136.204745] tfw_test: TEST_RUN(http2_parser, parses_enforce_ext_req)
[  136.205508] tfw_test: h2 req: 00 07 3a 6d 65 74 68 6f 64 03 47 45 54 00 07 3a 73 63 68 65 6d 65 05 68 74 74 70 73 00 05 3a 70  ..:method.GET..:scheme.https..:p
[  136.207709] tfw_test: h2 req: 00 07 3a 6d 65 74 68 6f 64 03 47 45 54 00 07 3a 73 63 68 65 6d 65 05 68 74 74 70 73 00 05 3a 70  ..:method.GET..:scheme.https..:p
[  136.209411] tfw_test: h2 req: 00 07 3a 6d 65 74 68 6f 64 03 47 45 54 00 07 3a 73 63 68 65 6d 65 05 68 74 74 70 73 00 05 3a 70  ..:method.GET..:scheme.https..:p
[  136.211270] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:315: error: 
[  136.212094] tfw_test: FAIL:
[  136.212657] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.213416] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com", strlen("natsys-lab.com"), 0))) => 0x0
[  136.214529] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:315: error: 
[  136.215333] tfw_test: FAIL:
[  136.215986] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.216731] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com", strlen("natsys-lab.com"), 0))) => 0x0
[  136.217775] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:315: error: 
[  136.218587] tfw_test: FAIL:
[  136.219126] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.219823] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com", strlen("natsys-lab.com"), 0))) => 0x0
[  136.220858] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:315: error: 
[  136.221574] tfw_test: FAIL:
[  136.222079] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.222752] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com", strlen("natsys-lab.com"), 0))) => 0x0
[  136.223726] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:315: error: 
[  136.224419] tfw_test: FAIL:
[  136.224890] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.225524] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com", strlen("natsys-lab.com"), 0))) => 0x0
[  136.226733] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:315: error: 
[  136.227414] tfw_test: FAIL:
[  136.227874] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.228497] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com", strlen("natsys-lab.com"), 0))) => 0x0
[  136.229443] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:315: error: 
[  136.230147] tfw_test: FAIL:
[  136.230594] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.231205] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com", strlen("natsys-lab.com"), 0))) => 0x0
[  136.232138] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:315: error: 
[  136.232795] tfw_test: FAIL:
[  136.233233] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.234128] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com", strlen("natsys-lab.com"), 0))) => 0x0
[  136.235553] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:315: error: 
[  136.236390] tfw_test: FAIL:
[  136.236816] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.237422] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com", strlen("natsys-lab.com"), 0))) => 0x0
[  136.238376] tfw_test: h2 req: 00 07 3a 6d 65 74 68 6f 64 03 47 45 54 00 07 3a 73 63 68 65 6d 65 05 68 74 74 70 73 00 05 3a 70  ..:method.GET..:scheme.https..:p
[  136.239730] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:330: error: 
[  136.240360] tfw_test: FAIL:
[  136.240778] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.241361] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com:8080", strlen("natsys-lab.com:8080"), 0))) => 0x0
[  136.242603] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:330: error: 
[  136.243236] tfw_test: FAIL:
[  136.243676] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.244251] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com:8080", strlen("natsys-lab.com:8080"), 0))) => 0x0
[  136.245456] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:330: error: 
[  136.246284] tfw_test: FAIL:
[  136.246729] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.247303] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com:8080", strlen("natsys-lab.com:8080"), 0))) => 0x0
[  136.248532] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:330: error: 
[  136.249191] tfw_test: FAIL:
[  136.249612] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.250558] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com:8080", strlen("natsys-lab.com:8080"), 0))) => 0x0
[  136.252307] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:330: error: 
[  136.252940] tfw_test: FAIL:
[  136.253375] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.253975] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com:8080", strlen("natsys-lab.com:8080"), 0))) => 0x0
[  136.255208] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:330: error: 
[  136.255918] tfw_test: FAIL:
[  136.256342] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.256937] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com:8080", strlen("natsys-lab.com:8080"), 0))) => 0x0
[  136.258188] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:330: error: 
[  136.258833] tfw_test: FAIL:
[  136.259259] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.259853] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com:8080", strlen("natsys-lab.com:8080"), 0))) => 0x0
[  136.261098] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:330: error: 
[  136.261761] tfw_test: FAIL:
[  136.262188] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.262783] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com:8080", strlen("natsys-lab.com:8080"), 0))) => 0x0
[  136.264038] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:330: error: 
[  136.264691] tfw_test: FAIL:
[  136.265119] tfw_test:   test__http2_parser__parses_enforce_ext_req()
[  136.265726] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com:8080", strlen("natsys-lab.com:8080"), 0))) => 0x0
[  136.267917] tfw_test: TEST_RUN(http2_parser, parses_enforce_ext_req_rmark)
[  136.268611] tfw_test: h2 req: 00 07 3a 6d 65 74 68 6f 64 03 47 45 54 00 07 3a 73 63 68 65 6d 65 05 68 74 74 70 73 00 05 3a 70  ..:method.GET..:scheme.https..:p
[  136.270098] tfw_test: h2 req: 00 07 3a 6d 65 74 68 6f 64 03 47 45 54 00 07 3a 73 63 68 65 6d 65 05 68 74 74 70 73 00 05 3a 70  ..:method.GET..:scheme.https..:p
[  136.271567] tfw_test: h2 req: 00 07 3a 6d 65 74 68 6f 64 03 47 45 54 00 07 3a 73 63 68 65 6d 65 05 68 74 74 70 73 00 05 3a 70  ..:method.GET..:scheme.https..:p
[  136.272966] tfw_test: h2 req: 00 07 3a 6d 65 74 68 6f 64 03 47 45 54 00 07 3a 73 63 68 65 6d 65 05 68 74 74 70 73 00 05 3a 70  ..:method.GET..:scheme.https..:p
[  136.274391] tfw_test: h2 req: 00 07 3a 6d 65 74 68 6f 64 03 47 45 54 00 07 3a 73 63 68 65 6d 65 05 68 74 74 70 73 00 05 3a 70  ..:method.GET..:scheme.https..:p
[  136.275891] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:414: error: 
[  136.276579] tfw_test: FAIL:
[  136.277027] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.277685] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.279017] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:414: error: 
[  136.279681] tfw_test: FAIL:
[  136.280128] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.280767] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.282601] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:414: error: 
[  136.283266] tfw_test: FAIL:
[  136.283717] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.284357] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.285718] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:414: error: 
[  136.286586] tfw_test: FAIL:
[  136.287035] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.287686] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.289017] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:414: error: 
[  136.289693] tfw_test: FAIL:
[  136.290142] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.290812] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.292185] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:414: error: 
[  136.292882] tfw_test: FAIL:
[  136.293329] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.293981] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.295363] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:414: error: 
[  136.296239] tfw_test: FAIL:
[  136.296697] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.297349] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.299539] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:414: error: 
[  136.300219] tfw_test: FAIL:
[  136.300676] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.301326] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.302697] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:414: error: 
[  136.303397] tfw_test: FAIL:
[  136.303853] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.304533] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.305922] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:414: error: 
[  136.306754] tfw_test: FAIL:
[  136.307189] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.307847] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.309184] tfw_test: h2 req: 00 07 3a 6d 65 74 68 6f 64 03 47 45 54 00 07 3a 73 63 68 65 6d 65 05 68 74 74 70 73 00 05 3a 70  ..:method.GET..:scheme.https..:p
[  136.310610] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:428: error: 
[  136.311279] tfw_test: FAIL:
[  136.311720] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.312358] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.313695] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:428: error: 
[  136.314612] tfw_test: FAIL:
[  136.315061] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.315752] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.317764] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:428: error: 
[  136.318426] tfw_test: FAIL:
[  136.318868] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.319504] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.320847] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:428: error: 
[  136.321512] tfw_test: FAIL:
[  136.321980] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.322648] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.324041] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:428: error: 
[  136.324752] tfw_test: FAIL:
[  136.325205] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.326040] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.327427] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:428: error: 
[  136.328207] tfw_test: FAIL:
[  136.328679] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.329318] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.331425] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:428: error: 
[  136.332077] tfw_test: FAIL:
[  136.332511] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.333138] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.334449] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:428: error: 
[  136.335099] tfw_test: FAIL:
[  136.335535] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.336358] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.337730] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:428: error: 
[  136.338381] tfw_test: FAIL:
[  136.338813] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.339439] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.340727] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:428: error: 
[  136.341372] tfw_test: FAIL:
[  136.341803] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.342446] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.343765] tfw_test: h2 req: 00 07 3a 6d 65 74 68 6f 64 03 47 45 54 00 07 3a 73 63 68 65 6d 65 05 68 74 74 70 73 00 05 3a 70  ..:method.GET..:scheme.https..:p
[  136.345146] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:442: error: 
[  136.345861] tfw_test: FAIL:
[  136.346583] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.347213] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.349070] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:442: error: 
[  136.349753] tfw_test: FAIL:
[  136.350193] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.350830] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.352152] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:442: error: 
[  136.352817] tfw_test: FAIL:
[  136.353298] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.354010] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.355418] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:442: error: 
[  136.356326] tfw_test: FAIL:
[  136.356822] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.357486] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.358822] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:442: error: 
[  136.359580] tfw_test: FAIL:
[  136.360040] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.360696] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.362410] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:442: error: 
[  136.363064] tfw_test: FAIL:
[  136.363498] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.364124] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.365539] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:442: error: 
[  136.366412] tfw_test: FAIL:
[  136.366846] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.367471] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.368768] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:442: error: 
[  136.369415] tfw_test: FAIL:
[  136.369850] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.370483] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.371823] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:442: error: 
[  136.372469] tfw_test: FAIL:
[  136.372902] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.373527] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
[  136.374821] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser.c:442: error: 
[  136.375465] tfw_test: FAIL:
[  136.376004] tfw_test:   test__http2_parser__parses_enforce_ext_req_rmark()
[  136.376644] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "natsys-lab.com" ":" "80", strlen("natsys-lab.com" ":" "80"), 0))) => 0x0
........
[  136.421536] tfw_test: TEST_RUN(http2_parser_hpack, static_table_all_indexes_for_request)
[  136.422215] tfw_test: h2 req: 82 87 84                                                                                         ...
[  136.423450] tfw_test: h2 req: 82 87 85                                                                                         ...
[  136.424679] tfw_test: h2 req: 82 87 85 0f 0d 01 31                                                                             ......1
[  136.426174] tfw_test: h2 req: 83 87 84                                                                                         ...
[  136.427697] tfw_test: h2 req: 82 87 84 01 09 6c 6f 63 61 6c 68 6f 73 74 0f 00 05 75 74 66 2d 38 90 0f 02 2c 66 72 2d 43 48 2c  .....localhost...utf-8...,fr-CH,
[  136.429064] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:499: error: 
[  136.429717] tfw_test: FAIL:
[  136.430090] tfw_test:   test__http2_parser_hpack__static_table_all_indexes_for_request()
[  136.430718] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.431576] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:499: error: 
[  136.432200] tfw_test: FAIL:
[  136.432597] tfw_test:   test__http2_parser_hpack__static_table_all_indexes_for_request()
[  136.433236] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.434114] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:499: error: 
[  136.434756] tfw_test: FAIL:
[  136.435132] tfw_test:   test__http2_parser_hpack__static_table_all_indexes_for_request()
[  136.435803] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.436692] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:499: error: 
[  136.437354] tfw_test: FAIL:
[  136.437753] tfw_test:   test__http2_parser_hpack__static_table_all_indexes_for_request()
[  136.438417] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.439325] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:499: error: 
[  136.439975] tfw_test: FAIL:
[  136.440359] tfw_test:   test__http2_parser_hpack__static_table_all_indexes_for_request()
[  136.441029] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.441911] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:499: error: 
[  136.442767] tfw_test: FAIL:
[  136.443155] tfw_test:   test__http2_parser_hpack__static_table_all_indexes_for_request()
[  136.443821] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.445235] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:499: error: 
[  136.446059] tfw_test: FAIL:
[  136.446445] tfw_test:   test__http2_parser_hpack__static_table_all_indexes_for_request()
[  136.447110] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.447995] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:499: error: 
[  136.448743] tfw_test: FAIL:
[  136.449138] tfw_test:   test__http2_parser_hpack__static_table_all_indexes_for_request()
[  136.449801] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.450812] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:499: error: 
[  136.451446] tfw_test: FAIL:
[  136.451834] tfw_test:   test__http2_parser_hpack__static_table_all_indexes_for_request()
[  136.452478] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.453326] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:499: error: 
[  136.453993] tfw_test: FAIL:
[  136.454388] tfw_test:   test__http2_parser_hpack__static_table_all_indexes_for_request()
[  136.455032] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.456080] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:499: error: 
[  136.456725] tfw_test: FAIL:
[  136.457129] tfw_test:   test__http2_parser_hpack__static_table_all_indexes_for_request()
[  136.457797] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.459449] tfw_test: h2 req: 83 87 84 0f 0d 01 37 0f 10 0a 74 65 78 74 2f 70 6c 61 69 6e 0f 14 0c 31 30 30 2d 63 6f 6e 74 69  ......7...text/plain...100-conti
[  136.460815] tfw_test: TEST_RUN(http2_parser_hpack, increment_all_static_indexes_for_request)
[  136.461500] tfw_test: h2 req: 82 47 05 68 74 74 70 73 44 01 2f 41 09 6c 6f 63 61 6c 68 6f 73 74 4f 05 75 74 66 2d 38 50 0d 67  .G.httpsD./A.localhostO.utf-8P.g
[  136.462880] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:667: error: 
[  136.463541] tfw_test: FAIL:
[  136.463957] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.464669] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.465599] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:667: error: 
[  136.466447] tfw_test: FAIL:
[  136.466893] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.467575] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.468465] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:667: error: 
[  136.469123] tfw_test: FAIL:
[  136.469541] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.470266] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.471226] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:667: error: 
[  136.471956] tfw_test: FAIL:
[  136.472409] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.473134] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.474587] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:667: error: 
[  136.475273] tfw_test: FAIL:
[  136.475711] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.476616] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.477696] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:667: error: 
[  136.478374] tfw_test: FAIL:
[  136.478824] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.479521] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.480441] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:667: error: 
[  136.481113] tfw_test: FAIL:
[  136.481538] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.482257] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.483167] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:667: error: 
[  136.483877] tfw_test: FAIL:
[  136.484328] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.485083] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.486191] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:667: error: 
[  136.486906] tfw_test: FAIL:
[  136.487407] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.488150] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.489128] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:667: error: 
[  136.489866] tfw_test: FAIL:
[  136.490638] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.491366] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.492872] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:667: error: 
[  136.493603] tfw_test: FAIL:
[  136.494046] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.494769] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.495690] tfw_test: h2 req: 82 d5 d4 d3 d2 d1 d0 cf ce cd cc cb ca c9 c8 c7 c6 c5 c4 c3 c2 c1 c0 bf be                       .........................
[  136.497472] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:787: error: 
[  136.498179] tfw_test: FAIL:
[  136.498602] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.499304] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.500206] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:787: error: 
[  136.500880] tfw_test: FAIL:
[  136.501313] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.502036] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.502952] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:787: error: 
[  136.503684] tfw_test: FAIL:
[  136.504126] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.504887] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.505841] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:787: error: 
[  136.506838] tfw_test: FAIL:
[  136.507267] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.507984] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.509432] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:787: error: 
[  136.510133] tfw_test: FAIL:
[  136.510576] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.511287] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.512201] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:787: error: 
[  136.512892] tfw_test: FAIL:
[  136.513325] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.514047] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
[  136.514963] tfw_test: /root/tempesta/fw/t/unit/test_http2_parser_hpack.c:787: error: 
[  136.515680] tfw_test: FAIL:
[  136.516331] tfw_test:   test__http2_parser_hpack__increment_all_static_indexes_for_request()
[  136.517058] tfw_test:   EXPECT_TRUE((tfw_str_eq_cstr(&req->host, "localhost", strlen("localhost"), 0))) => 0x0
.......
[  136.583481] tfw_test: TEST_RUN(http_match, choose_host)
[  136.583906] tfw_test: /root/tempesta/fw/t/unit/test_http_match.c:829: error: 
[  136.584427] tfw_test: FAIL:
[  136.584737] tfw_test:   test__http_match__choose_host()
[  136.585161] tfw_test:   EXPECT_EQ((2), (match_id)) => (0x2, 0xffffffffffffffff)
[  136.585692] tfw_test: /root/tempesta/fw/t/unit/test_http_match.c:849: error: 
[  136.586611] tfw_test: FAIL:
[  136.586959] tfw_test:   test__http_match__choose_host()
[  136.587398] tfw_test:   EXPECT_EQ((3), (match_id)) => (0x3, 0xffffffffffffffff)
[  136.587913] tfw_test: /root/tempesta/fw/t/unit/test_http_match.c:862: error: 
[  136.588414] tfw_test: FAIL:
[  136.588712] tfw_test:   test__http_match__choose_host()
[  136.589126] tfw_test:   EXPECT_EQ((2), (match_id)) => (0x2, 0xffffffffffffffff)
.....
[  136.605699] tfw_test: finish - failed 115 assertions
Tests failed!
make: *** [Makefile:168: test] Error 1
```

Reply:
Yes it is so after https://github.com/tempesta-tech/tempesta/commit/7670e8776ba564af7d1f1c62eda7cdcf54c8ea93

Reply:
Fixed in the last commit